### PR TITLE
Bundle flag and balance checks into authorize

### PR DESCRIPTION
**Pull request by gregtatcam**

### CODE REVIEW

File: src/test/jtx/impl/mpt.cpp [hunk 1]
```diff
@@ -161,7 +161,24 @@ MPTTester::authorize(MPTAuthorize const& arg)
     }
     if (arg.holder)
         jv[sfMPTokenHolder.jsonName] = arg.holder->human();
-    submit(arg, jv);
+    if (submit(arg, jv) == tesSUCCESS)
+    {
+        if (arg.account == nullptr || *arg.account == issuer_)
```
Comment (shawnxie999) [inline src/test/jtx/impl/mpt.cpp:166]: whats the need for `arg.account == nullptr` check here?

Comment (shawnxie999) [reply src/test/jtx/impl/mpt.cpp:166]: Nevermind, the default account is the issuer.

**Review by shawnxie999 — APPROVED** [review]
Please feel free to merge it. Thanks!

File: src/test/jtx/impl/mpt.cpp (hunk 1 above)
Comment (gregtatcam) [reply src/test/jtx/impl/mpt.cpp:166]: It's valid for `arg.account` to be nullptr. If `arg.account` is nullptr then by default it's the `issuer_`.

**Pull request merged**